src/main/java/com/kodilla/ecommercee/mapper/UserMapper.java

```diff
@@ -0,0 +1,41 @@
+package com.kodilla.ecommercee.mapper;
+
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.dto.UserDto;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserMapper {
+
+    public User mapToUser(final UserDto userDto) {
+        return new User(
+                userDto.getUserId(),
+                userDto.getUsername(),
+                userDto.isStatus(),
+                userDto.getUserKey(),
+                userDto.getCart(),
+                userDto.getOrders()
```

Comment:
Are you sure it works? `userDto.getOrders()`? I don't see changes for `UserDto` - there should be `List<OrderDto>` not `List<Order>`, as Andrzej said we shouldn't mixed domain-objects and dto-objects in one class.

Reply:
corrected